src/journal/journal-file.c

```diff
@@ -464,6 +497,9 @@ static int journal_file_refresh_header(JournalFile *f) {
         /* Sync the online state to disk */
         (void) fsync(f->fd);
 
+        /* We likely just created a new file, also sync the directory this file is located in. */
+        (void) fsync_directory_of_file(f->fd);
```

Comment:
Why not just make `journal_file_open()` either:
1. stick the containing directory's fd in `JournalFile.dir_fd` or
2. stick the containing directory's path in `JournalFile.dir_path`

It may make more sense to simply change `joural_file_open()`'s api to something like `journal_file_openat()` and require the caller to supply the containing directory's fd, then we just take ownership of the containing directory fd in `JournalFile` in addition to the journal file's fd.

There are potential races and correctness issues in what's going on with this patch as-is.  Depending on what's being done at the filesystem, you could have an open journal fd to a file that is no longer in the directory found at that path, if for instance an admin fucked around removing and recreating things in var.

If instead the flow for opening journals was to open the containing directory, then open/create the journal relative to the containing fd, and the containing fd and journal fd were supplied to journal_file_open() as a pair then we can be certain to do the right thing, regardless of what's been done to the names in the filesystem in the mean time.